rm-datasource/src/main/java/org/apache/seata/rm/datasource/xa/ConnectionProxyXA.java

```diff
@@ -127,10 +130,12 @@ private void xaEnd(XAXid xaXid, int flags) throws XAException {
      * @param applicationData application data
      * @throws SQLException SQLException
      */
-    public synchronized void xaCommit(String xid, long branchId, String applicationData) throws XAException {
-        XAXid xaXid = XAXidBuilder.build(xid, branchId);
-        xaResource.commit(xaXid, false);
-        releaseIfNecessary();
+    public void xaCommit(String xid, long branchId, String applicationData) throws XAException {
```

Comment:
Is there a more elegant way to write code? resourceLock.obtain() is more complex than the synchronized keyword.

Reply:
我认为这点还是可以接受的，因为
I think this is still acceptable because
```
synchronized(this){
}
```
与
with
```
try(ResourceLock ignored = resourceLock.obtain()){
}
```
书写方式区别不大。
The writing style is not much different.

Reply:
> Is there a more elegant way to write code? resourceLock.obtain() is more complex than the synchronized keyword.

The reason i write this way is to avoid having to add unlock to finally;
`
class X {
    private final ReentrantLock lock = new ReentrantLock();
    public void m() {
        lock.lock();  // block until condition holds
        try {
            // ... method body
        } finally {
            lock.unlock();//unlock
        }
    }
}
`